apps/web/src/services/useActivitiesAllowedByOccupation.ts

```diff
@@ -0,0 +1,114 @@
+import { useCallback, useEffect, useState } from 'react';
+import { API_ENDPOINT } from '../common';
+import { useHttp } from './useHttp';
+import { AllowedActivityByOccupation } from '../common/interfaces';
+import { PageOptions } from 'src/components/Pagination';
+import {
+  OccupationalScopeOfPracticeSortKeys,
+  PaginationRO,
+  Permissions,
+  SortOrder,
+} from '@tbcm/common';
+
+const DEFAULT_PAGE_SIZE = 5;
+
+export const useActivitiesAllowedByOccupation = (occupationId?: string) => {
+  const { fetchData, isLoading } = useHttp();
+  const [allowedActivities, setAllowedActivities] = useState<AllowedActivityByOccupation[]>([]);
+  const [pageSize, setPageSize] = useState(DEFAULT_PAGE_SIZE);
+  const [pageIndex, setPageIndex] = useState(1);
+  const [total, setTotal] = useState(0);
+  const [sortKey, setSortKey] = useState<OccupationalScopeOfPracticeSortKeys>();
+  const [sortOrder, setSortOrder] = useState<SortOrder>();
+  const [searchText, setSearchText] = useState('');
+  const [filterByPermission, setFilterByPermission] = useState<Permissions>();
+
+  const onPageOptionsChange = ({ pageIndex: pgIndex, pageSize: size }: PageOptions) => {
+    if (size !== pageSize) {
+      setPageSize(size);
+      setPageIndex(1);
+    } else {
+      setPageIndex(pgIndex);
+    }
+  };
+
+  const nextSortOrder = useCallback((order?: SortOrder) => {
+    switch (order) {
+      case SortOrder.ASC:
+        return SortOrder.DESC;
+      case SortOrder.DESC:
+        return undefined;
```

Comment:
Nit - Having undefined here feels like a bit of a code smell. 
It might be worth refactoring to add an unordered state

Reply:
Thanks. I will address it separately. Created https://eydscanada.atlassian.net/browse/TBCM-151.